src/server/db_slice.cc

```diff
@@ -27,6 +27,10 @@ ABSL_FLAG(uint32_t, max_segment_to_consider, 4,
           "The maximum number of dashtable segments to scan in each eviction "
           "when heartbeat based eviction is triggered under memory pressure.");
 
+ABSL_FLAG(double, oom_capacity_ratio, 1.1,
+          "Return OOM if the capacity of the table after growing multiplied by the ratio "
```

Comment:
`Prevents table from growing if number of free slots x average object size x this ratio is larger than memory budget`